src/java/org/apache/cassandra/db/compaction/TimeOrderedKeyCompactionStrategyOptions.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.compaction;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.utils.Pair;
+
+import java.util.Map;
+
+final class TimeOrderedKeyCompactionStrategyOptions
+{
+    protected static final String SPLIT_FACTOR_KEY = "split_factor";
+    protected static final int DEFAULT_SPLIT_FACTOR = 8;
+
+    // format: "1230,40.0" : 1230MB or 40%
+    // default: "int_max:20" : 20% of the keys are tombstoned
+    protected static final String DEFAULT_GARBAGE_SIZE_THRESHOLD_STR = Integer.MAX_VALUE + ",20";
+    protected static final Pair<Long, Double> DEFAULT_GARBAGE_SIZE_THRESHOLD = parseSizeThreshold(DEFAULT_GARBAGE_SIZE_THRESHOLD_STR);
+    protected static final String WINDOW_GARBAGE_SIZE_THRESHOLD_KEY = "window_garbage_size_threshold";
+    protected static final String GLOBAL_GARBAGE_SIZE_THRESHOLD_KEY = "global_garbage_size_threshold";
+
+    final TimeWindowCompactionStrategyOptions twcsOptions;
+
+    final long windowCompactionSizeInMB;
+    final double windowCompactionSizePercent;
+
+    final long windowCompactionGlobalSizeInMB;
+    final double windowCompactionGlobalSizePercent;
+
+    final int splitFactor;
```

Comment:
What's the significant of splitFactor?

From what I can see, it is used eventually by TBSCWriter. But I thought the idea was to split SSTables aligning with time window boundaries, and this will just split them into equal time ranges.

Reply:
Its a heuristic. It still needs to be seen how it will perform in actual workload. 
Though the idea is that, splitting partially first may lead to compaction of one of the split tombstones first. I am relying on the fact that tombstones wont uniformly distributed across time range. hence splitting by some factor, i am looking for that time range which is loaded with tombstones, which if other constraint succeed, will lead to compaction. Other files dont need to be split if they dont have enough garbage.